cmd/kubectl-gadget/utils/trace.go

```diff
@@ -290,32 +450,160 @@ RetryLoop:
-		printTraceFeedback(contextLogger.Errorf, nodeErrors)
+		printTraceFeedback(printErrorFunction, nodeErrors)
 
 		if successNodeCount == 0 {
-			deleteTraces(contextLogger, traceRestClient, traceID)
-			contextLogger.Fatalf("Failed to run the gadget on all nodes: None of them succeeded")
+			return errors.New("Failed to run the gadget on all nodes: None of them succeeded")
 		}
+
 		break RetryLoop
 	}
 
+	client, err := k8sutil.NewClientsetFromConfigFlags(KubernetesConfigFlags)
+	if err != nil {
+		return fmt.Errorf("Error in creating setting up Kubernetes client: %w", err)
+	}
+
 	if customResultsDisplay == nil {
-		if outputMode != "Stream" {
-			panic(fmt.Errorf("OutputMode=%q needs a custom display function", outputMode))
+		// If we arrived here, it means we have at least one successful node, so it
+		// is safe to use trace pointer.
+		if trace.Spec.OutputMode != "Stream" {
+			return fmt.Errorf("OutputMode=%q needs a custom display function", trace.Spec.OutputMode)
+		}
+
+		params := &CommonFlags{
+			OutputMode: config.CommonFlags.OutputMode,
+			Node:       config.CommonFlags.Node,
+			Verbose:    config.CommonFlags.Verbose,
 		}
-		genericStreamsDisplay(contextLogger, client, params, &results, transformLine)
+
+		genericStreamsDisplay(client, params, &traces, transformLine)
 	} else {
-		customResultsDisplay(contextLogger, nodes, &results)
+		nodes, err := client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return fmt.Errorf("Error listing nodes: %w", err)
+		}
+
+		customResultsDisplay(nodes, &traces)
+	}
+
+	return nil
+}
+
+// DeleteTrace deletes the traces for the given trace ID using RESTClient.
+func DeleteTrace(traceID string) error {
+	traceRestClient, err := getRestClient()
+	if err != nil {
+		return err
 	}
-	deleteTraces(contextLogger, traceRestClient, traceID)
+
+	deleteTraces(traceRestClient, traceID)
+
+	return nil
+}
+
+// getTraceListFromParameters returns traces associated with the given config.
+func getTraceListFromParameters(config *TraceConfig) ([]gadgetv1alpha1.Trace, error) {
+	namespace := ""
+	if !config.CommonFlags.AllNamespaces {
+		namespace = GetNamespace()
+	}
+
+	// If some of the arguments are empty, they will just be ignored when
+	// searching for corresponding trace.
+	labels := fmt.Sprintf("gadgetName=%v,nodeName=%v,namespace=%v,podName=%v,containerName=%v,outputMode=%v",
+		config.GadgetName, config.CommonFlags.Node, namespace, config.CommonFlags.Podname, config.CommonFlags.Containername, config.TraceOutputMode)
+
+	var listTracesOptions = metav1.ListOptions{
+		LabelSelector: labels,
+		FieldSelector: fields.Everything().String(),
+	}
+
+	traces, err := getTraceListFromOptions(listTracesOptions)
+	if err != nil {
+		return []gadgetv1alpha1.Trace{}, err
+	}
+
+	return traces.Items, nil
+}
+
+// PrintAllTraces prints all traces corresponding to the given config.CommonFlags.
+func PrintAllTraces(config *TraceConfig) error {
```

Comment:
I am not sure I understand what is the purpose of giving the possibility to pass this config here. I expect that it would allow me to filter the traces I want to be displayed, e.g. if I just want to see the traces in the `demo` namespace and so on. However, from what I see, this function does not work like that. Instead, it will show you the traces that match **exactly** the filter you use. For instance, let's start the biolatency gadget to create a trace:

```
$ ./kubectl-gadget biolatency start --node worker
Successfully created trace with ID: "AvN9nL9H4Q7hkxuR"!
You will need to use this ID to report trace.
In case you lost it, you can list existing traces with 'list' subcommand
```

However, I was expecting that the following command would show me all the traces in all the nodes and all the namespaces but it does not: 
```
$ ./kubectl-gadget biolatency list -A
NAMESPACE    POD    CONTAINER    TRACEID
```

Same here:
```
$ ./kubectl-gadget biolatency list --node worker -A
NAMESPACE    POD    CONTAINER    TRACEID
```

I will be only able to see my trace if the filter is identical:
```
$ ./kubectl-gadget biolatency list --node worker
NAMESPACE    POD    CONTAINER    TRACEID
default                          AvN9nL9H4Q7hkxuR
```

In my opinion, we should just show all the biolatency traces and that's it. What do you think? 

cc @mauriciovasquezbernal